### DISCUSSION TITLE
How to create the core.remote for chatops!

### DISCUSSION THREAD

**Discussion by BathiniRavi** [StackStorm/st2#5443] [ChatOps]
I need to add sudo credentials in that file but not sure how to add that or is there any core.remote_sudo file at chatops level ?
+++++++++++++++++++++++++++++++
# packs/my-chatops/aliases/remote.yaml
---
name: "remote_shell_cmd"
action_ref: "core.remote"
description: "Execute a command on a remote host via SSH."
formats:
  - "run {{cmd}} on {{hosts}}"
  
PS - I'm running Stackstorm as Docker container.

Also the above sample is not working in my case, maybe my configs are wrong !
Below is the output from my Bot -

!help response from bot
![image](https://user-images.githubusercontent.com/55383317/142019888-1d5cedfc-4504-4ada-8826-34db47070672.png)]

command response from bot
(![image](https://user-images.githubusercontent.com/55383317/142020067-dce041e5-5474-4ceb-bdca-a4f538ef4111.png))

 configs from container
![image](https://user-images.githubusercontent.com/55383317/142021827-dec8c15c-da69-4e5d-aefb-1de8ff44887c.png)

**Comment by join240j** [comment]
if you want use with ssh keys you need change "run {{cmd}} on {{username}}@{{hosts}}" 
for run = !st2 run "date" on username@10.0.0.1
if you want use with username and password conection on telegram u need change "run {{cmd}} on {{username}}@{{hosts}} with password {{password}}"
for run  = !st2 run "date" on "username@10.0.0.1" with password "password"

**Comment by nzlosh** [comment]
From a security point of view, posting passwords as cleartext into a chat service will leak the password in chat history and possibly through logging by the bot/stackstorm/chat provider.  Exposing `core.remote` with arbitrary commands as an action-alias is bad practice because it opens a very simple attack vector to execute any command as any user on all machines that StackStorm has access to.
